proto/vtctldata.proto

```diff
@@ -209,16 +209,25 @@ message Shard {
   topodata.Shard shard = 3;
 }
 
+enum ShardedAutoIncrementHandling {
+  LEAVE = 0;
+  REMOVE = 1;
+  REPLACE = 2;
+}
+
 message WorkflowOptions {
   string tenant_id = 1;
   // Remove auto_increment clauses on tables when moving them to a sharded
-  // keyspace.
-  bool strip_sharded_auto_increment = 2;
+  // keyspace and optionally replace them with vschema AutoIncrement
+  // definitions.
+  ShardedAutoIncrementHandling strip_sharded_auto_increment = 2;
```

Comment:
We should rename this to `sharded_auto_increment_handling` or similar. Renaming proto fields does not break compatibility in anyway.

Reply:
Btw, when did we add this field? If it was within v21, we are fine. otherwise we'd have to reserve this and add a new field because we are changing the type.

Reply:
Good point on renaming. Will do. That field was added in v20.

Reply:
then changing bool to enum is theoretically a breaking change. BUT, since this is between vtctldclient and vtctld, we can assume that their versions should match and not worry about reserving the field.

Reply:
The potential problem with renaming the proto field is that the field name is used when marshalling/unmarshalling the field to protojson or prototext. So it would mean that we do not read the value properly for old/existing workflows ... which is fine because in the older versions any actions taken from that value are done when creating the workflow and the value is not needed/used after that.

Reply:
I tried to address this here: https://github.com/vitessio/vitess/pull/16860/commits/733198b0c7bbc7368a5ce007b28a87890fbb0f2e